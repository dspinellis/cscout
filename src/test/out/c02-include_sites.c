Table: Ids
 EID  NAME            READONLY  UNDEFMACRO  MACRO  MACROARG  ORDINARY  SUETAG  SUMEMBER  LABEL  TYPEDEF  ENUM   YACC   FUN    CSCOPE  LSCOPE  UNUSED
 127  MAXDIGIT        0     0       1   0     0     0   0     0  0    0  0  0  0   0   1
 165  used            0     0       0  0     1      0   0     0  0    0  0  0  0   1    1
 317  FILE            0     0       1   0     0     0   0     0  0    0  0  0  0   0   0
 463  x               0     0       0  0     1      0   0     0  0    0  0  0  0   1    1
 617  main            1      0       0  0     1      0   0     0  0    0  0  1   0   1    0
 799  qqq             0     0       0  0     1      0   0     0  0    0  0  0  0   1    1
1247  foo             0     0       0  0     1      0   0     0  0    0  0  1   0   1    1
4163  __DATE__        1      0       1   0     0     0   0     0  0    0  0  0  0   0   1
4611  __TIME__        1      0       1   0     0     0   0     0  0    0  0  0  0   0   1
5003  __FILE__        1      0       1   0     0     0   0     0  0    0  0  0  0   0   1
5409  __LINE__        1      0       1   0     0     0   0     0  0    0  0  0  0   0   1
5675  __STDC__        1      0       1   0     0     0   0     0  0    0  0  0  0   0   1
7495  _cscout_dummy1  1      0       0  0     1      0   0     0  0    0  0  1   1    0   0
Table: Tokens
FID  FOFFSET   EID
  2      297  4163
  2      329  4611
  2      357  5003
  2      386  5409
  2      405  5675
  2      471   617
  2      535  7495
  2      558  7495
  4        7   107
  4       22   317
  4       55   317
  4       92   317
  5       11   165
  7        8   127
  7       32   463
  7       43   617
  7       56   799
  7       88  1247
Table: Rest
FID  FOFFSET  CODE
  2      287  \u0000a\u0000a#define 
  2      305    
  2      320  \u0000a#define 
  2      337   
  2      348  \u0000a#define 
  2      365   
  2      377  \u0000a#define 
  2      394   1\u0000a#define 
  2      413   1\u0000a\u0000a
  2      466  \u0000aint 
  2      475  ();\u0000a
  2      522  \u0000astatic void 
  2      549  (void) { 
  2      572  (); }\u0000a
  3      152  \u0000a\u0000a\u0000a#pragma includepath 
  3      195  \u0000a
  3      239  \u0000astatic void _cscout_dummy2(void) { _cscout_dummy2(); }\u0000a
  4        0  #ifdef 
  4       11  \u0000d\u0000a#define  
  4       26   
  4       38  \u0000d\u0000a#else\u0000d\u0000a#define 
  4       59   
  4       71  \u0000d\u0000a#endif\u0000d\u0000a\u0000d\u0000a#include 
  4       96  \u0000d\u0000a\u0000d\u0000a
  5        0  extern int 
  5       15  ;\u0000d\u0000a
  7        0  #define 
  7       16   11\u0000d\u0000aextern int 
  7       33  ;\u0000d\u0000aextern 
  7       47  ();\u0000d\u0000aint 
  7       59  ;\u0000d\u0000a\u0000d\u0000a
  7       86  \u0000d\u0000a
  7       91  () {\u0000d\u0000a	
  7      116  }\u0000d\u0000a\u0000d\u0000a
Table: Projects
PID  NAME
 16  unspecified
 17  Prj1
 18  Prj2
Table: IdProj
 EID  PID
 165   17
 317   17
 617   17
4163   17
4611   17
5003   17
5409   17
5675   17
7495   17
 127   18
 317   18
 463   18
 617   18
 799   18
1247   18
4163   18
4611   18
5003   18
5409   18
5675   18
7495   18
Table: Files
FID  NAME                 RO     NCHAR  NCCOMMENT  NSPACE  NLCOMMENT  NBCOMMENT  NLINE  MAXLINELEN  NSTRING  NULINE  NPPDIRECTIVE  NPPCOND  NPPFMACRO  NPPOMACRO  NPPTOKEN  NCTOKEN  NCOPIES  NSTATEMENT  NPFUNCTION  NFFUNCTION  NPVAR  NFVAR  NAGGREGATE  NAMEMBER  NENUM  NEMEMBER  NINCFILE
  2  host-defs.h          1     578        367      29          0          3     22          61        3       0             5        0          0          5        37       18        1           1           0           1      0      0           0         0      0         0         0
  3  host-incs.h          1     295        187      13          0          2     13          54        1       0             1        0          0          0        16        0        1           0           0           0      0      0           0         0      0         0         0
  4  c02-include_sites.c  0    100          0      23          0          0      8          26        2       0             6        1          0          1        19        1        1           0           0           0      0      0           0         0      0         0         1
  5  unused2.h            0     18          0       4          0          0      1          17        0       0             0        0          0          0         4        4        1           0           0           0      1      0           0         0      0         0         0
  6  unused1.h            0      0          0       0          0          0      0           0        0       0             0        0          0          0         0        0        1           0           0           0      0      0           0         0      0         0         0
  7  prj2.c               0    121         34      26          2          0     11          21        0       0             1        0          0          1        21       18        1           0           1           0      2      0           0         0      0         0         0
Table: FileProj
FID  PID
  2   17
  3   17
  4   17
  5   17
  1   18
  2   18
  3   18
  4   18
  6   18
  7   18
Table: Definers
PID  CUID  BASEFILEID  DEFINERID
 18     4           4          1
 18     7           7          2
Table: Includers
PID  CUID  BASEFILEID  INCLUDERID
 17     2           2           1
 17     4           3           1
 17     4           4           1
 17     4           5           4
 18     2           2           1
 18     2           2           1
 18     4           3           1
 18     4           4           1
 18     4           6           4
 18     7           3           1
 18     7           7           1
Table: Providers
PID  CUID  PROVIDERID
 17     2           2
 18     2           2
 18     2           2
 18     7           7
Table: IncTriggers
PID  CUID  BASEFILEID  DEFINERID  FOFFSET  LEN
 18     7           7          2      471    4
Table: Functions
  ID  NAME            ISMACRO  DEFINED  DECLARED  FILESCOPED  FID  FOFFSET  FANIN
1247  foo             0    1     1      0         7       88      0
6599  main            0    0    1      0         2      471      0
7495  _cscout_dummy1  0    1     1      1          2      535      1
Table: FunctionMetrics
FUNCTIONID  NCHAR  NCCOMMENT  NSPACE  NLCOMMENT  NBCOMMENT  NLINE  MAXLINELEN  NSTRING  NULINE  NPPDIRECTIVE  NPPCOND  NPPFMACRO  NPPOMACRO  NPPTOKEN  NCTOKEN  NSTMT  NOP  NUOP  NNCONST  NCLIT  NIF  NELSE  NSWITCH  NCASE  NDEFAULT  NBREAK  NFOR  NWHILE  NDO  NCONTINUE  NGOTO  NRETURN  NPID  NFID  NMID  NID  NUPID  NUFID  NUMID  NUID  NGNSOC  NPARAM  MAXNEST  NLABEL  FANIN  FANOUT  CCYCL1  CCYCL2  CCYCL3  CSTRUC   CHAL  IFLOW  FIDBEGIN  FOFFSETBEGIN  FIDEND  FOFFSETEND
      1247     23         15       4          1          0      2          18        0       0             0        0          0          0         1        1      0    0     0        0      0    0      0        0      0         0       0     0       0    0          0      0        0     0     0     0    0      0      0      0     0      16       0        0       0      0       0       1       1       1   0.0  0.0  0.0         7            95       7         117
      7495     21          0       3          0          0      1          20        0       0             0        0          0          0         5        5      1    0     0        0      0    0      0        0      0         0       0     0       0    0          0      0        0     0     1     0    1      0      1      0     1       8       0        0       0      1       1       1       1       1   1.0  0.0  1.0         2           557       2         577
Table: FunctionId
FUNCTIONID  ORDINAL   EID
      1247        0  1247
      6599        0   617
      7495        0  7495
Table: Fcalls
SOURCEID  DESTID
    7495    7495
Done
