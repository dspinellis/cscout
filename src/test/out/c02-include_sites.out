Table: Ids
EID	NAME	READONLY	UNDEFMACRO	MACRO	FUNMACRO	MACROARG	CPPCONST	CPPSTRVAL	DEFCCONSTVAL	NOTDEFCCONSTVAL	EXPCCONSTVAL	NOTEXPCCONSTVAL	ORDINARY	SUETAG	SUMEMBER	LABEL	TYPEDEF	ENUM	YACC	FUN	CSCOPE	LSCOPE	UNUSED
127	MAXDIGIT	0	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
165	used	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
303	FILE	0	0	1	0	0	1	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0
449	x	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
589	main	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
757	qqq	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
1149	foo	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
4163	__DATE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
4611	__TIME__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
5003	__FILE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
5409	__LINE__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
5675	__STDC__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
7495	_cscout_dummy1	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	1	0	0
Table: Tokens
FID	FOFFSET	EID
2	297	4163
2	329	4611
2	357	5003
2	386	5409
2	405	5675
2	471	589
2	535	7495
2	558	7495
4	7	107
4	21	303
4	52	303
4	86	303
5	11	165
7	8	127
7	31	449
7	41	589
7	53	757
7	81	1149
Table: Rest
FID	FOFFSET	CODE
2	287	\u0000a\u0000a#define 
2	305	  
2	320	\u0000a#define 
2	337	 
2	348	\u0000a#define 
2	365	 
2	377	\u0000a#define 
2	394	 1\u0000a#define 
2	413	 1\u0000a\u0000a
2	466	\u0000aint 
2	475	();\u0000a
2	522	\u0000astatic void 
2	549	(void) { 
2	572	(); }\u0000a
3	152	\u0000a\u0000a\u0000a#pragma includepath 
3	195	\u0000a
3	239	\u0000astatic void _cscout_dummy2(void) { _cscout_dummy2(); }\u0000a
4	0	#ifdef 
4	11	\u0000a#define  
4	25	 
4	37	\u0000a#else\u0000a#define 
4	56	 
4	68	\u0000a#endif\u0000a\u0000a#include 
4	90	\u0000a\u0000a
5	0	extern int 
5	15	;\u0000a
7	0	#define 
7	16	 11\u0000aextern int 
7	32	;\u0000aextern 
7	45	();\u0000aint 
7	56	;\u0000a\u0000a
7	80	\u0000a
7	84	() {\u0000a	
7	107	}\u0000a\u0000a
Table: Projects
PID	NAME
23	unspecified
24	Prj1
25	Prj2
Table: IdProj
EID	PID
165	24
303	24
589	24
4163	24
4611	24
5003	24
5409	24
5675	24
7495	24
127	25
303	25
449	25
589	25
757	25
1149	25
4163	25
4611	25
5003	25
5409	25
5675	25
7495	25
Table: Files
FID	NAME	RO
2	host-defs.h	1
3	host-incs.h	1
4	c02-include_sites.c	0
5	unused2.h	0
6	unused1.h	0
7	prj2.c	0
Table: Filemetrics
FID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NCOPIES	NINCFILE	NPFUNCTION	NFFUNCTION	NPVAR	NFVAR	NAGGREGATE	NAMEMBER	NENUM	NEMEMBER
2	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	16	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	2	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	1	0	0	0	0	0	0
2	1	548	367	29	0	3	22	61	47	NULL	1	1	0	0	0	5	0	0	5	37	2	0	0	2	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	2	5	8	1	1	5	7	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
3	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	15	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
3	1	243	187	13	0	2	13	48	28	NULL	1	1	0	0	0	1	0	0	0	16	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
4	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
4	1	69	0	15	0	0	8	21	0	NULL	0	0	0	0	0	6	1	0	1	19	0	0	0	0	0	2	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	4	0	0	2	2	0	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
5	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	6	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	4	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	1	0	0	0	0	0
5	1	11	0	3	0	0	1	10	6	NULL	0	0	0	0	0	0	0	0	0	4	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	1	1	0	0	1	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
6	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
6	1	0	0	0	0	0	0	0	0	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
7	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	7	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	3	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	NULL	4	4	0	NULL	4	0	NULL	NULL	NULL	1	0	2	0	0	0	0	0
7	1	95	32	17	2	0	11	20	13	NULL	1	1	0	0	0	1	0	0	1	21	3	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	1	5	4	0	1	5	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
Table: FileProj
FID	PID
2	24
3	24
4	24
5	24
1	25
2	25
3	25
4	25
6	25
7	25
Table: LineProj
FID	PID	LNUM
2	24	1
2	24	2
2	24	3
2	24	4
2	24	5
2	24	6
2	24	7
2	24	8
2	24	9
2	24	10
2	24	11
2	24	12
2	24	13
2	24	14
2	24	15
2	24	16
2	24	17
2	24	18
2	24	19
2	24	20
2	24	21
2	24	22
3	24	1
3	24	2
3	24	3
3	24	4
3	24	5
3	24	6
3	24	7
3	24	8
3	24	9
3	24	10
3	24	11
3	24	12
3	24	13
4	24	1
4	24	4
4	24	5
4	24	6
4	24	7
4	24	8
5	24	1
2	25	1
2	25	2
2	25	3
2	25	4
2	25	5
2	25	6
2	25	7
2	25	8
2	25	9
2	25	10
2	25	11
2	25	12
2	25	13
2	25	14
2	25	15
2	25	16
2	25	17
2	25	18
2	25	19
2	25	20
2	25	21
2	25	22
3	25	1
3	25	2
3	25	3
3	25	4
3	25	5
3	25	6
3	25	7
3	25	8
3	25	9
3	25	10
3	25	11
3	25	12
3	25	13
4	25	1
4	25	2
4	25	3
4	25	6
4	25	7
4	25	8
7	25	1
7	25	2
7	25	3
7	25	4
7	25	5
7	25	6
7	25	7
7	25	8
7	25	9
7	25	10
7	25	11
Table: LinePos
FID	FOFFSET	LNUM
2	0	1
2	4	2
2	51	3
2	54	4
2	100	5
2	119	6
2	122	7
2	184	8
2	231	9
2	234	10
2	284	11
2	288	12
2	289	13
2	321	14
2	349	15
2	378	16
2	397	17
2	416	18
2	417	19
2	467	20
2	479	21
2	523	22
3	0	1
3	4	2
3	51	3
3	54	4
3	78	5
3	97	6
3	100	7
3	149	8
3	153	9
3	154	10
3	155	11
3	196	12
3	240	13
4	0	1
4	12	2
4	38	3
4	44	4
4	69	5
4	76	6
4	77	7
4	91	8
5	0	1
7	0	1
7	20	2
7	34	3
7	49	4
7	58	5
7	59	6
7	80	7
7	81	8
7	89	9
7	107	10
7	109	11
Table: Definers
PID	CUID	BASEFILEID	DEFINERID
25	4	4	1
25	7	7	2
Table: Includers
PID	CUID	BASEFILEID	INCLUDERID
24	2	2	1
24	4	3	1
24	4	4	1
24	4	5	4
25	2	2	1
25	2	2	1
25	4	3	1
25	4	4	1
25	4	6	4
25	7	3	1
25	7	7	1
Table: Providers
PID	CUID	PROVIDERID
24	2	2
25	2	2
25	2	2
25	7	7
Table: IncTriggers
PID	CUID	BASEFILEID	DEFINERID	FOFFSET	LEN
25	7	7	2	471	4
Table: Functions
ID	NAME	ISMACRO	DEFINED	DECLARED	FILESCOPED	FID	FOFFSET	FANIN
1149	foo	0	1	1	0	7	81	0
6599	main	0	0	1	0	2	471	0
7495	_cscout_dummy1	0	1	1	1	2	535	1
Table: FunctionDefs
FUNCTIONID	FIDBEGIN	FOFFSETBEGIN	FIDEND	FOFFSETEND
1149	7	88	7	108
7495	2	557	2	577
Table: FunctionMetrics
FUNCTIONID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NGNSOC	NMPARAM	NFPARAM	NEPARAM	FANIN	FANOUT	CCYCL1	CCYCL2	CCYCL3	CSTRUC	CHAL	IFLOW
1149	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	16	NULL	0	NULL	0	0	1	1	1	0.0	0.0	0.0
1149	1	21	14	3	1	0	3	17	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	0	0	1	1	1	0.0	0.0	0.0
7495	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	NULL	1	0	1	NULL	1	0	NULL	8	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
7495	1	22	0	3	0	0	1	21	5	NULL	0	1	0	0	0	0	0	0	0	5	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	1	0	1	0	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
Table: FunctionId
FUNCTIONID	ORDINAL	FID	FOFFSET
1149	0	7	81
6599	0	2	471
7495	0	2	535
Table: Fcalls
SOURCEID	DESTID
7495	7495
Done
